packages/test/src/test-workflows.ts

```diff
@@ -353,6 +356,14 @@ test('successString', async (t) => {
   compareCompletion(t, req, makeSuccess([makeCompleteWorkflowExecution(defaultPayloadConverter.toPayload('success'))]));
 });
 
+test('continueAsNewSuggested', async (t) => {
```

Comment:
Can you get an integration test for this please instead of this unit test?
I'm considering removing (most of) these unit tests.

Reply:
@bergundy I suggested a UT on that one because we can't make assumptions on when will the server reports `suggestContinueAsNew = true`. Sure, we could run for a long time and expect that, at some point, the flag will turn from false to true, but that seems more suitable for the feature test repo.

Reply:
+1. Unit tests are very useful, specifically in situations like this which James gets at.

Reply:
We had a discussion about this. The system test is just a loop with Sleep(1) so that's easy... However, the current threshold is 4K activations, and against a cloud target that's probably 6-7 minutes just for that test. Also, this is a recent change, not sure if it has made it to the cli server yet. And then, the whole discussion on whether this is the right place to check that the server behaves... 

BTW, I don't think that `features` should become the system test repo either, I like the fact that is trying to be simple and focus on one "feature" at a time, rather than stressing  end-to-end with a combination of "features"...

Reply:
You could use a workflow that does a bunch of search attribute upserts.

Reply:
Integration tests are bound to test some of the server behavior. But without them you can't be sure that the feature actually works E2E.

Reply:
So each upsert creates a different activation but we can batch many in one round-trip, is that what you mean?

I'm a big fan of E2E testing, the only issue is whether it belongs to this repo or not...

Reply:
Fair enough, there's no use arguing this over and over. Let's just add an integration test. However, I would _also_ keep that unit test, because it has a clear, well defined semantic, which E2E test won't have on that specific property.

@antlai-temporal The current default thresholds for the server to suggest CAN is 4096 events or 4MB ([see here](https://github.com/temporalio/temporal/blob/548c1506bf831cf836a6232d86a118bc89477308/service/history/configs/config.go#L475C1-L478)). You can do multiple commands per WFT to grow history very quickly, eg.: if you start 512 timers concurrently of 1ms each, on the next WFT, history will have grown by ~1024 events (TimerStarted and TimerFired). History should reach the threshold very quickly. Now, don't assume that server will actually raise the flag at that exact value. Simply continue looping until either the flag turns true (success) or the history reaches some really unreasonable length (eg. > 40K events), at which point you can fail.

Reply:
I added the system test that you described. 
I also added another one that checks historySizeBytes is visible in WorkflowExecutionInfo